homeassistant/components/sws12500/sensor.py

```diff
@@ -0,0 +1,269 @@
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="",
+    ),
+    WeatherSensorEntityDescription(
+        key=RAIN,
+        name="Rain",
+        native_unit_of_measurement=UnitOfPrecipitationDepth.INCHES,
+        icon="",
+        device_class=SensorDeviceClass.PRECIPITATION,
+        suggested_unit_of_measurement=UnitOfPrecipitationDepth.MILLIMETERS,
+    ),
+    WeatherSensorEntityDescription(
+        key=DAILY_RAIN,
+        name="Daily precipitation",
+        native_unit_of_measurement="in/d",
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="",
+        device_class=SensorDeviceClass.PRECIPITATION_INTENSITY,
+        suggested_unit_of_measurement="mm/h",
+    ),
+    WeatherSensorEntityDescription(
+        key=SOLAR_RADIATION,
+        name="Solar irradiance",
+        native_unit_of_measurement=UnitOfIrradiance.WATTS_PER_SQUARE_METER,
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="",
+        device_class=SensorDeviceClass.IRRADIANCE,
+    ),
+    WeatherSensorEntityDescription(
+        key=UV,
+        name="UV index",
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="",
+        native_unit_of_measurement="",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info=None,
+) -> None:
+    """Set up Weather Station sensors."""
+
+    coordinator: WeatherDataUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+    sensors = []
+    for description in SENSOR_TYPES:
+        sensors.append(WeatherSensor(hass, description, coordinator))
+    async_add_entities(sensors)
+
+
+class WeatherSensor(
+    CoordinatorEntity[WeatherDataUpdateCoordinator], RestoreEntity, SensorEntity
+):
+    """Implementation of Weather Sensor entity."""
+
+    entity_description: WeatherSensorEntityDescription
+    _attr_has_entity_name = True
+    _attr_should_poll = False
+
+    def __init__(
+        self,
+        hass,
+        description: WeatherSensorEntityDescription,
+        coordinator: WeatherDataUpdateCoordinator,
+    ) -> None:
+        """Initialize sensor."""
+        super().__init__(coordinator)
+        self.hass = hass
+        self.coordinator = coordinator
+        self.entity_description = description
+        self._state: StateType = None
+        self._attrs: dict[str, Any] = {}
+        self._available = False
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._state = self.coordinator.data.get(self.entity_description.key)  # type: ignore[attr-defined]
+        if not self.available and self._state:
+            self._available = True
+
+        self.async_write_ha_state()
+        self.async_registry_entry_updated()
+
+    @property
+    def translation_key(self):
+        """Return translation key."""
+        return self.entity_description.translation_key
```

Comment:
Why rewrite it if the same is already inherited from HA `Entity` class through the `SensorEntity` class anyway?

The same is for `name`, `device_class`, `icon` and more which do exactly the same as the default HA behaviour